core/planner.cpp

```diff
@@ -450,20 +450,23 @@ static struct gaschanges *analyze_gaslist(struct diveplan *diveplan, struct dive
 }
 
 /* sort all the stops into one ordered list */
-static int *sort_stops(int *dstops, int dnr, struct gaschanges *gstops, int gnr)
+static std::vector<int> sort_stops(int dstops[], size_t dnr, std::vector<gaschanges> gstops)
 {
-	int i, gi, di;
-	int total = dnr + gnr;
-	int *stoplevels = malloc(total * sizeof(int));
+	int total = dnr + gstops.size();
+	std::vector<int> stoplevels(total);
+
+	/* Can't happen. */
+	if (dnr == 0)
+		return std::vector<int>();
 
 	/* no gaschanges */
-	if (gnr == 0) {
-		memcpy(stoplevels, dstops, dnr * sizeof(int));
+	if (gstops.empty()) {
+		std::copy(dstops, dstops + dnr, stoplevels.begin());
 		return stoplevels;
 	}
-	i = total - 1;
-	gi = gnr - 1;
-	di = dnr - 1;
+	int i = static_cast<int>(total) - 1;
```

Comment:
Because `size_t` is unsigned and that would make the loop `while (i >= 0)` an infinite loop.

Unsigned integers are evil and `size_t` was, in hindsight, a terrible design decision. Though understandable in the context of 32 bit systems.